packages/tpetra/core/test/ImportExport2/ImportExport2_UnitTests.cpp

```diff
@@ -222,6 +235,271 @@ namespace {
+  // {
+  //   using Teuchos::VERB_EXTREME;
+  //   using Teuchos::VERB_NONE;
+  //   typedef Tpetra::global_size_t GST;
+  //   typedef double Scalar;
+
+  //   const GST INVALID = Teuchos::OrdinalTraits<GST>::invalid ();
+  //   const RCP<const Comm<int> > comm = getDefaultComm();
+  //   const int numProcs = comm->getSize();
+  //   const int myRank = comm->getRank();
+   
+  //   // Prepare for verbose output, if applicable.
+  //   Teuchos::EVerbosityLevel verbLevel = verbose ? VERB_EXTREME : VERB_NONE;
+  //   const bool doPrint = includesVerbLevel (verbLevel, VERB_EXTREME, true);
+  //   if (doPrint) 
+  //     {
+  //     out << "relayGIDOwner unit test" << endl;
+  //   }
+  //   OSTab tab1 (out); // Add one tab level
+
+  //   std::ostringstream err;
+  //   int lclErr = 0;
+
+  //   RCP<ParameterList> crsMatPlist = getCrsMatrixParameterList ();
+  //   const LO num_local_elements = 3;
+  //   RCP<const Map<LO, GO> >my_map = createContigMap<LO, GO> (INVALID, num_local_elements, comm);
+                                   
+  //   RCP<CrsMatrix<Scalar, LO, GO> > mat =
+  //     rcp (new CrsMatrix<Scalar, LO, GO> (my_map, 1, StaticProfile, crsMatPlist));
+   
+  //   // Create a simple diagonal source graph.
+  //   for (GO globalrow = my_map->getMinGlobalIndex();
+  //        globalrow <= my_map->getMaxGlobalIndex();
+  //        ++globalrow) {
+  //     mat->insertGlobalValues (globalrow,
+  //                              tuple<GO> (globalrow),
+  //                              tuple<Scalar> (globalrow));
+  //   }
+    
+  //   mat->fillComplete ();
+
+  //   auto MDM = mat->getDomainMap();
+  //   Teuchos::Array<int> ProcsFrom;
+  //   for(int i=0;i<numProcs;++i) if(i!=myRank) ProcsFrom.push_back(i);
+  //   Teuchos::Array<std::pair<int,GO> > vpg;
+  //   // we can compute gids/pid 
+  //   GO g=0;
+  //   for(int i=0;i<numProcs;++i) {
+  //     vpg.push_back(std::pair<int,GO> (i,g++));
+  //     vpg.push_back(std::pair<int,GO> (i,g++));
+  //     vpg.push_back(std::pair<int,GO> (i,g++));
+  //   }
+
+  //   Teuchos::Array<std::tuple<int,int,GO > > relayedGIDs;
+  //   Tpetra::Import_Util::relayGIDOwner (MDM,  vpg,  ProcsFrom,  relayedGIDs);
+
+  //   // now test that everyone sent us the same stuff. 
+  //   for(auto && p: relayedGIDs) {
+  //     GO g;
+  //     int pidowner;
+  //     int pidrelayed;
+  //     std::tie(pidrelayed,pidowner,g) = p;
+  //     if(pidrelayed == myRank)
+  //       lclErr++;
+      
+  //   }
+
+  //   int gblErr=-1;
+  //   reduceAll<int, int> (*comm, REDUCE_MAX, lclErr, outArg (gblErr));
+  //   if (gblErr != 0) {
+  //     for (int r = 0; r < numProcs; ++r) {
+  //       if (r == myRank) {
+  //         cerr << err.str () << endl;
+  //       }
+  //       comm->barrier ();
+  //       comm->barrier ();
+  //       comm->barrier ();
+  //     }
+  //     TEUCHOS_TEST_FOR_EXCEPTION(true, std::logic_error, "Test failed!");
+  //   }
+
+  // TEST_EQUALITY(gblErr,0);
+  // }
```

Comment:
Why comment out this test?

Reply:
The test got munged by a merge, I've restored it from devel.